README.md

```diff
@@ -1,2 +1,119 @@
-# as-helm-chart
-Access Server Helm chart
+![openvpn-as](https://as-prod-public.s3.us-west-1.amazonaws.com/openvpn_logo.svg)
+# OpenVPN Access Server Helm chart
+
+## Introduction
+
+This chart installs `openvpn/openvpn-as` Docker Hub image on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
+
+## Prerequisites
+
+- Kubernetes 1.20+
+- Helm 3.0+
+
+## Installing the Chart
+
+To install the chart with the release name `my-vpn`:
+
+```console
+helm install my-vpn ./as-helm-chart
+```
+
+These commands deploy openvpn-as on the Kubernetes cluster in the default configuration.
+
+## Uninstalling the Chart
+
+To uninstall/delete the `my-vpn` deployment:
+
+```console
+helm delete my-vpn
+```
+
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## Configuration
+
+The configurable parameters of the openvpn-as chart can be found in `values.yaml`.
+
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`.
+
+Alternatively, a YAML file that specifies the values for the parameters can be provided while installing the chart. For example,
+
+```console
+helm install my-vpn -f values.yaml ./as-helm-chart
+```
+
+## Access server custom configuration
+
+Custom configuration is possible via Kubernetes [postStart](https://kubernetes.io/docs/tasks/configure-pod-container/attach-handler-lifecycle-event/) feature.
+
+***Important: If you plan to use `sacli` in the `postStart` script, ensure that the Access Server service is fully up and running beforehand.***  
+You can find an example of how to wait for the service in the Inline Script below.
+
+If enabled, this chart runs default `scripts/configure.sh` from the chart.
+This is just an example which waits while AS initialization is finished.
+The default `scripts/configure.sh` can be overridden.
+
+### Override with Inline Script
+
+#### Add to values.yaml:
+
+```console
+postStart:
+  customScript: |
+    #!/bin/bash
+    # Waiting for AS service initialization
+    until /usr/local/openvpn_as/scripts/sacli status 2>/dev/null |grep -q '"api": "on"'
+    do
+        sleep 2
+    done
+    /usr/local/openvpn_as/scripts/sacli --user "openvpn" --new_pass "secure123" SetLocalPassword
+```
+
+#### Use External Secret
```

Comment:
Any specific reason you called this "External" secret? "External Secret" makes me think of https://external-secrets.io/, but you seem to just refer to a normal Kubernetes secret. What is "external" about it?

Reply:
Ah, I understand it now. You create a secret from the script, so you call it external secret if the user provides it and you do not create it yourself. Maybe some explanation might help:

"The chart will create a Kubernetes secret out of the provided script. Instead you can also provide your own secret directly with the postStart.externalSecretName key. Note that the the script must be available as configure.sh in the secret."